src/pages/product/Product.js

```diff
@@ -1,17 +1,44 @@
-import React from 'react'
+import React,{useReducer} from 'react'
 import { Filter, ProductCard } from 'component'
 import './product.css'
 import '../../component/card/productcard.css'
 import { useProduct } from 'context/ProductContext'
+import { reducer } from 'reducer/filter-reducer'
+import { getPriceWishFilter } from 'filterFuntion/sort'
+import { categoryFiltered } from 'filterFuntion/filter-category'
+import { ratingFilter } from 'filterFuntion/ratingFilter'
 
 export const Product = () => {
   const { product } = useProduct()
+  const [state, dispatch] = useReducer(reducer, {
+    sortBy: ""  ,
+    Sports: false,
+    Casual: false,
+    Formal: false,
+    Sneaker: false,
+    rating: false
+  });
```

Comment:
i guess here it can be made an array of objects so that we can loop over it using 2d array to compare for category name in products db with the category name.
Just a suggestion .

Reply:
ok i will refactor in future